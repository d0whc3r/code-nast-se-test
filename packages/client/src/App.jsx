import News from './components/News/News'

function App() {
  return (
    <>
      <header>
        <h1 className="text-center">News board</h1>
      </header>
      <main className="md:container md:mx-auto">
        <News />
      </main>
    </>
  )
}

export default App
