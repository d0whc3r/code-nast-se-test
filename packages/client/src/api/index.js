function generateApiUrl(path, params) {
  const url = new URL(`/api${path}`, import.meta.env.VITE_API_URL)
  if (params) {
    Object.entries(params).forEach(([key, param]) => {
      if (param) {
        url.searchParams.append(key, param)
      }
    })
  }
  return url.toString()
}

export function getNews({ topic = '', ...rest } = {}) {
  return fetch(generateApiUrl(`/news/${topic}`, rest))
    .then(async (response) => {
      const result = response.json()
      if (response.ok) {
        return result
      }
      throw await result
    })
    .catch((error) => {
      console.log(error)
      throw error
    })
}
