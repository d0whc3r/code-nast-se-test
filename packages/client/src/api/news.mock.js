export const mockedNews = {
  'status': 'ok',
  'totalResults': 22170,
  'articles': [{
    'source': { 'id': null, 'name': 'Enriquedans.com' },
    'author': 'Enrique Dans',
    'title': 'Rusia y las VPN como síntoma',
    'description': 'Las descargas y el uso de redes privadas virtuales (VPNs) en Rusia se han disparado desde el inicio de la brutal invasión de Ucrania, con niveles que se aproximan al medio millón de descargas diario frente a las alrededor de diez mil que había antes de que Vl…',
    'url': 'https://www.enriquedans.com/2022/05/rusia-y-las-vpn-como-sintoma.html',
    'urlToImage': 'https://www.enriquedans.com/wp-content/uploads/2022/05/VPN.jpeg',
    'publishedAt': '2022-05-07T14:46:50Z',
    'content': 'Las descargas y el uso de redes privadas virtuales (VPNs) en Rusia se han disparado desde el inicio de la brutal invasión de Ucrania, con niveles que se aproximan al medio millón de descargas diario … [+3209 chars]',
  }, {
    'source': { 'id': null, 'name': 'Microsiervos.com' },
    'author': 'wicho@microsiervos.com (Wicho)',
    'title': 'Rusia hace el primer lanzamiento orbital de un cohete Angará 1.2 tras 25 años de desarrollo',
    'description': 'En la noche del 29 de abril de 2022 Rusia hizo por fin el primer lanzamiento orbital de un cohete Angará 1.2 después de unos 25 años de desarrollo en los que hubo varias idas y venidas en cuanto a la financiación y al diseño del cohete. De hecho es posible qu…',
    'url': 'https://www.microsiervos.com/archivo/espacio/primer-lanzamiento-orbital-cohete-angara-1-2.html',
    'urlToImage': 'https://img.youtube.com/vi/TBEV3foZMKE/maxresdefault.jpg',
    'publishedAt': '2022-04-30T11:24:34Z',
    'content': 'Por @Wicho 30 de Abril de 2022\r\nEn la noche del 29 de abril de 2022 Rusia hizo por fin el primer lanzamiento orbital de un cohete Angará 1.2 después de unos 25 años de desarrollo en los que hubo vari… [+2292 chars]',
  }, {
    'source': { 'id': null, 'name': 'Genbeta.com' },
    'author': 'Bárbara Bécares',
    'title': 'Este vídeo del presidente de Ucrania diciendo que esnifa cocaína es un bulo que ha usado tecnología para cambiar el original',
    'description': 'Ha comenzado a circular un vídeo por redes sociales del presidente de Ucrania, Volodimir Zelenski, afirmando que consume cocaína, que esta sustancia es el “mejor energizante” y que recomienda su uso a “todo el mundo”. En realidad, es un fake que ha usado tecn…',
    'url': 'https://www.genbeta.com/actualidad/este-video-presidente-ucrania-diciendo-que-esnifa-cocaina-bulo-que-ha-usado-tecnologia-para-cambiar-original',
    'urlToImage': 'https://i.blogs.es/f8fe53/1366_2000/840_560.jpeg',
    'publishedAt': '2022-04-27T12:40:57Z',
    'content': 'Ha comenzado a circular un vídeo por redes sociales del presidente de Ucrania, Volodimir Zelenski, afirmando que consume cocaína, que esta sustancia es el mejor energizante y que recomienda su uso a … [+2261 chars]',
  }, {
    'source': { 'id': null, 'name': 'Hipertextual' },
    'author': 'Gabriel Erard',
    'title': 'El rebrote de la COVID-19 en China golpea con fuerza a la industria de los chips',
    'description': 'El rebrote de la COVID-19 en China ha provocado la implementación de duras medidas restrictivas en algunas de sus ciudades más importantes. Shanghái es una de las urbes que más las padece, generando gran preocupación entre las compañías tecnológicas que opera…',
    'url': 'https://hipertextual.com/2022/04/rebrote-covid-19-china-chips',
    'urlToImage': 'https://hipertextual.com/wp-content/uploads/2021/09/cpu-chip-processor-computer.jpeg',
    'publishedAt': '2022-04-18T19:49:06Z',
    'content': 'El rebrote de la COVID-19 en China ha provocado la implementación de duras medidas restrictivas en algunas de sus ciudades más importantes. Shanghái es una de las urbes que más las padece, generando … [+3145 chars]',
  }, {
    'source': { 'id': null, 'name': 'Hipertextual' },
    'author': 'Gabriel Erard',
    'title': 'CD Projekt RED calma las aguas por el nuevo retraso de ‘The Witcher 3’ para PS5 y Xbox Series X|S',
    'description': 'El más reciente retraso que ha sufrido el lanzamiento de The Witcher 3 para PS5 y Xbox Series X|S ha obligado a CD Projekt RED a salir a bajarle el tono a la noticia. En una llamada con inversores (vía Polygon), el estudio polaco aseguró que el juego no ha ca…',
    'url': 'https://hipertextual.com/2022/04/the-witcher-3-resta-importancia-nueva-demora-next-gen',
    'urlToImage': 'https://hipertextual.com/wp-content/uploads/2019/12/hipertextual-nuevos-juegos-the-witcher-horizonte-cd-projekt-red-sapkowski-hacen-paces-2019203367.jpg',
    'publishedAt': '2022-04-18T21:50:00Z',
    'content': 'El más reciente retraso que ha sufrido el lanzamiento de The Witcher 3 para PS5 y Xbox Series X|S ha obligado a CD Projekt RED a salir a bajarle el tono a la noticia. En una llamada con inversores (v… [+2933 chars]',
  }, {
    'source': { 'id': null, 'name': 'Genbeta.com' },
    'author': 'Marcos Merino',
    'title': 'Vetado el uso en Rusia de Times New Roman, Arial, Verdana y otras fuentes, por culpa de las sanciones (no, no son software libre)',
    'description': 'La mayoría de equipos informáticos del mundo, sean Windows, Mac o Linux, incluyen un \'paquete básico\' de tipografías preinstaladas, que a todos nos sonarán: Times New Roman, Arial, Helvetica, Verdana, Tahoma… tipografías que la mayoría de los usuarios cree qu…',
    'url': 'https://www.genbeta.com/actualidad/vetado-uso-rusia-times-new-roman-arial-verdana-otras-fuentes-culpa-sanciones-no-no-software-libre',
    'urlToImage': 'https://i.blogs.es/27cfb9/fuentes/840_560.jpg',
    'publishedAt': '2022-04-15T10:29:30Z',
    'content': 'La mayoría de equipos informáticos del mundo, sean Windows, Mac o Linux, incluyen un \'paquete básico\' de tipografías preinstaladas, que a todos nos sonarán: Times New Roman, Arial, Helvetica, Verdana… [+3079 chars]',
  }, {
    'source': { 'id': null, 'name': 'Hipertextual' },
    'author': 'Hipertextual (Redacción)',
    'title': 'Canadá aprueba una ley para condenar los crímenes que ocurran en la Luna',
    'description': 'Canadá suele ser noticia por muy pocas cosas. Pero en este caso, el tranquilo y apacible país norteamericano salta a la palestra por la aprobación de una ley que permitirá enjuiciar los crímenes cometidos en la Luna. También en el resto del espacio, la Estaci…',
    'url': 'https://hipertextual.com/2022/05/canada-condena-crimenes-luna',
    'urlToImage': 'https://i0.wp.com/hipertextual.com/wp-content/uploads/2022/03/astronauta-gravity-espacio.jpg?fit=2200%2C1237&quality=50&strip=all&ssl=1',
    'publishedAt': '2022-05-06T12:30:00Z',
    'content': 'Canadá suele ser noticia por muy pocas cosas. Pero en este caso, el tranquilo y apacible país norteamericano salta a la palestra por la aprobación de una ley que permitirá enjuiciar los crímenes come… [+3194 chars]',
  }, {
    'source': { 'id': null, 'name': 'Xataka.com' },
    'author': 'Carlos Prego',
    'title': 'Taiwán recibe el 95% de su Internet por cables submarinos vulnerables ante China. Ahora teme quedarse aislada',
    'description': 'La guerra de Ucrania está teniendo un efecto peculiar en el Pacífico, a miles de kilómetros de Kiev: ha avivado el recelo con el que Taiwán ve su frágil conexión a Internet. Más que con la infraestructura de su Red, que también, el miedo de la isla asiática s…',
    'url': 'https://www.xataka.com/otros/taiwan-recibe-95-su-internet-cables-submarinos-vulnerables-china-ahora-teme-quedarse-aislada',
    'urlToImage': 'https://i.blogs.es/b287cd/1366_2000-1-/840_560.jpg',
    'publishedAt': '2022-04-23T18:50:57Z',
    'content': 'La guerra de Ucrania está teniendo un efecto peculiar en el Pacífico, a miles de kilómetros de Kiev: ha avivado el recelo con el que Taiwán ve su frágil conexión a Internet. Más que con la infraestru… [+4821 chars]',
  }, {
    'source': { 'id': null, 'name': 'Xataka.com' },
    'author': 'Javier Marquez',
    'title': 'Netflix pierde suscriptores por primera vez en una década y se desploma en la bolsa',
    'description': 'Año tras año vimos como Netflix sumaba nuevos usuarios, pese a la creciente competencia, pero esa tendencia se ha detenido. Después del frenazo de 2021, la compañía ha anunciado una pérdida de 200.000 suscriptores en el primer trimestre de 2022 y, por si eso …',
    'url': 'https://www.xataka.com/empresas-y-economia/netflix-pierde-suscriptores-primera-vez-decada-se-desploma-bolsa',
    'urlToImage': 'https://i.blogs.es/7fdcf4/netflix/840_560.jpeg',
    'publishedAt': '2022-04-19T22:24:45Z',
    'content': 'Año tras año vimos como Netflix sumaba nuevos usuarios, pese a la creciente competencia, pero esa tendencia se ha detenido. Después del frenazo de 2021, la compañía ha anunciadouna pérdida de 200.000… [+2643 chars]',
  }, {
    'source': { 'id': null, 'name': 'Genbeta.com' },
    'author': 'Marcos Merino',
    'title': 'El bulo sobre el reciente despixelado de las bases rusas en Google Maps permite detectar una escuadrilla de aviones de gomaespuma',
    'description': 'Es de conocimiento público que Google Maps censura habitualmente, desenfocándolas, ciertas instalaciones gubernamentales (sobre todo las de tipo militar) cuando aparecen en sus mapas. El Acuartalamiento Capitán Sevillano de Pozuelo de Alarcón (Madrid) es un e…',
    'url': 'https://www.genbeta.com/actualidad/bulo-reciente-despixelado-bases-rusas-google-maps-permite-detectar-escuadrilla-aviones-gomaespuma',
    'urlToImage': 'https://i.blogs.es/1e3b4d/rusos/840_560.jpg',
    'publishedAt': '2022-04-19T19:33:46Z',
    'content': 'Es de conocimiento público que Google Maps censura habitualmente, desenfocándolas, ciertas instalaciones gubernamentales (sobre todo las de tipo militar) cuando aparecen en sus mapas. El Acuartalamie… [+3198 chars]',
  }, {
    'source': { 'id': null, 'name': 'Genbeta.com' },
    'author': 'Bárbara Bécares',
    'title': 'Las criptomonedas están cayendo enormemente desde finales de año. Y su dependencia a las decisiones políticas es evidente',
    'description': 'Hay diferentes noticias de los últimos meses que apuntan a que el auge de los criptoactivos podríá haber llegado a su tope. El Banco Central Europeo ha decretado el fin de los tipos de interés a cero (que ayudaban a mantener la deuda pública controlada) para …',
    'url': 'https://www.genbeta.com/actualidad/criptomonedas-estan-cayendo-enormemente-finales-ano-su-dependencia-a-decisiones-politicas-evidente',
    'urlToImage': 'https://i.blogs.es/9ee576/1366_2000/840_560.jpeg',
    'publishedAt': '2022-04-29T12:23:03Z',
    'content': 'Hay diferentes noticias de los últimos meses que apuntan a que el auge de los criptoactivos podríá haber llegado a su tope. El Banco Central Europeo ha decretado el fin de los tipos de interés a cero… [+3261 chars]',
  }, {
    'source': { 'id': null, 'name': 'Genbeta.com' },
    'author': 'Bárbara Bécares',
    'title': 'Tras el bitcoin, la bancarrota: El Salvador fue el primer país en adoptarlo y ahora vive las consecuencias de su gran caída',
    'description': 'El Salvador fue el primer país del mundo en adoptar el bitcoin como una moneda de curso legal. Hace unos días la República Centroafricana hizo lo mismo, aunque la decisión apunta más a una decisión del gobierno para agradar a Rusia, su gran apoyo, más que por…',
    'url': 'https://www.genbeta.com/actualidad/bitcoin-bancarrota-salvador-fue-primer-pais-adoptarlo-ahora-vive-consecuencias-su-gran-caida',
    'urlToImage': 'https://i.blogs.es/6c59d6/1366_2000/840_560.jpg',
    'publishedAt': '2022-05-13T09:46:22Z',
    'content': 'El Salvador fue el primer país del mundo en adoptar el bitcoin como una moneda de curso legal. Hace unos días la República Centroafricana hizo lo mismo, aunque la decisión apunta más a una decisión d… [+2800 chars]',
  }, {
    'source': { 'id': null, 'name': 'Muyinteresante.es' },
    'author': 'Sarah Romero',
    'title': 'Rusia lanza una nave espacial militar secreta en órbita alrededor de la Tierra',
    'description': 'El cohete Angará 1.2 se ha enviado tras 25 años en desarrollo. ¿Cuál es su función? ¿Podría usarse en la guerra contra Ucrania?',
    'url': 'https://www.muyinteresante.es/ciencia/articulo/rusia-lanza-una-nave-espacial-militar-secreta-en-orbita-alrededor-de-la-tierra-601651560217',
    'urlToImage': 'https://estaticos.muyinteresante.es/uploads/images/article/6270cf7a5cafe83ce49b92a1/dealwithit_0.jpg',
    'publishedAt': '2022-05-03T06:51:00Z',
    'content': 'Rusia continúa con sus lanzamientos espaciales a pesar de las continuas cancelaciones espaciales (y no espaciales) a nivel internacional por el inicio de la invasión de Ucrania por la Rusia de Vladim… [+1629 chars]',
  }, {
    'source': { 'id': null, 'name': 'Muyinteresante.es' },
    'author': 'Sarah Romero',
    'title': 'La ESA no volverá a colaborar con Rusia en futuras misiones a la Luna',
    'description': 'Tras la invasión de Ucrania por parte de Vladimir Putin, la Agencia Espacial Europea ha tomado una decisión.',
    'url': 'https://www.muyinteresante.es/ciencia/articulo/la-esa-no-volvera-a-colaborar-con-rusia-en-futuras-misiones-a-la-luna-511649948346',
    'urlToImage': 'https://estaticos.muyinteresante.es/uploads/images/article/625836795cafe8f75b22723d/luna-desaparecio.jpg',
    'publishedAt': '2022-04-14T15:08:49Z',
    'content': 'La Agencia Espacial Europea ha puesto fin, definitivamente, a su colaboración con Rusia en futuras misiones lunares. No volverán a cooperar con la agencia espacial Roscosmos de Moscú en las misiones … [+1641 chars]',
  }, {
    'source': { 'id': null, 'name': 'Hipertextual' },
    'author': 'Nicolás Rivera',
    'title': '¿Se acabó el baile para Netflix?',
    'description': 'Thibault Penin en Unsplash \r\n\n" data-medium-file="https://i0.wp.com/hipertextual.com/wp-content/uploads/2022/02/thibault-penin-awol7qqsffm-unsplash-scaled.jpeg?fit=800%2C534&quality=50&strip=all&ssl=1" data-large-file="https://i0.wp.com/hipertextual.com/wp-co…',
    'url': 'https://hipertextual.com/2022/04/se-acabo-el-baile-para-netflix',
    'urlToImage': 'https://hipertextual.com/wp-content/uploads/2022/02/thibault-penin-awol7qqsffm-unsplash-scaled.jpeg',
    'publishedAt': '2022-04-20T15:43:13Z',
    'content': 'Por primera vez en una década, Netflix ha perdido suscriptores durante el primer trimestre de 2022. Concretamente, han sido 200.000. Y, lo más preocupante de todo, es que la compañía proyecta perder … [+9027 chars]',
  }, {
    'source': { 'id': 'el-mundo', 'name': 'El Mundo' },
    'author': 'XAVIER COLÁS',
    'title': 'Rusia avisa de "medidas militares" ante la entrada de Finlandia en la OTAN',
    'description': 'Rusia se verá obligada a tomar medidas militares y de otra índole ante el ingreso de Suecia y Finlandia en la OTAN, advirtió el Ministerio de Exteriores de Rusia. El Kremlin...',
    'url': 'https://www.elmundo.es/internacional/2022/05/12/627cfad6fc6c8369208b4584.html',
    'urlToImage': 'https://phantom-elmundo.unidadeditorial.es/8e76ff989b2fada7368f5b9f244eb5ae/resize/1200/f/jpg/assets/multimedia/imagenes/2022/05/12/16523576029531.jpg',
    'publishedAt': '2022-05-12T13:36:51Z',
    'content': 'Rusia se verá obligada a tomar medidas militares y de otra índole ante el ingreso de Suecia y Finlandia en la OTAN, advirtió el Ministerio de Exteriores de Rusia. El Kremlin indicó que la reacción de… [+4115 chars]',
  }, {
    'source': { 'id': null, 'name': 'Xataka.com' },
    'author': 'Juan Carlos López',
    'title': 'Reino Unido quiere construir ahora un reactor nuclear al año. Es mucho más difícil de lo que parece',
    'description': 'La coyuntura actual está cambiando las reglas del juego. Durante los últimos años hemos presenciado cómo algunos países europeos, entre los que se encuentran España y Alemania, apuestan por un modelo energético que deja completamente de lado la energía nuclea…',
    'url': 'https://www.xataka.com/energia/reino-unido-quiere-construir-ahora-reactor-nuclear-al-ano-mucho-dificil-que-parece',
    'urlToImage': 'https://i.blogs.es/7ff8b7/reino-unido-centrales-nucleares/840_560.jpeg',
    'publishedAt': '2022-05-03T15:01:15Z',
    'content': 'La coyuntura actual está cambiando las reglas del juego. Durante los últimos años hemos presenciado cómo algunos países europeos, entre los que se encuentran España y Alemania, apuestan por un modelo… [+5965 chars]',
  }, {
    'source': { 'id': null, 'name': 'Marketingdirecto.com' },
    'author': 'Esther Lastra',
    'title': 'Henkel, la enésima multinacional que suspende sus operaciones en Rusia',
    'description': 'Henkel ha anunciado este martes su decisión de abandonar Rusia a la luz del actual desarrollo de los acontecimientos en Ucrania.\nLa entrada Henkel, la enésima multinacional que suspende sus operaciones en Rusia se publicó primero en Marketing Directo.',
    'url': 'https://www.marketingdirecto.com/anunciantes-general/anunciantes/henkel-enesima-multinacional-suspende-operaciones-rusia',
    'urlToImage': 'https://www.marketingdirecto.com/wp-content/uploads/2022/04/henkel.jpg',
    'publishedAt': '2022-04-19T15:49:17Z',
    'content': 'Henkel ha anunciado este martes su decisión de abandonar Rusia a la luz del actual desarrollo de los acontecimientos en Ucrania.Desde que la guerra de Ucrania estallara hace ya casi dos meses toda un… [+2230 chars]',
  }, {
    'source': { 'id': null, 'name': 'Xataka.com' },
    'author': 'Alberto de la Torre',
    'title': 'Y pensábamos que la gasolina estaba cara: Europa quiere dejar de importar petróleo ruso a finales de año',
    'description': 'La invasión rusa de Ucrania sigue adelante. Después de muchas otras sanciones y de eternas discusiones entre los países que conforman la Unión Europea, la presidenta de la Comisión Europea, Ursula von der Leyen, ha propuesto un veto progresivo al petróleo rus…',
    'url': 'https://www.xataka.com/movilidad/pensabamos-que-gasolina-estaba-cara-europa-quiere-dejar-importar-petroleo-ruso-a-finales-ano',
    'urlToImage': 'https://i.blogs.es/35b498/refineria-buena/840_560.jpg',
    'publishedAt': '2022-05-04T12:03:23Z',
    'content': 'La invasión rusa de Ucrania sigue adelante. Después de muchas otras sanciones y de eternas discusiones entre los países que conforman la Unión Europea, la presidenta de la Comisión Europea, Ursula vo… [+3387 chars]',
  }, {
    'source': { 'id': null, 'name': 'Muyinteresante.es' },
    'author': 'Muy Interesante',
    'title': '¿Quién está detrás de los ciberataques a Ucrania?',
    'description': 'El país también está inmerso en una guerra en la red.',
    'url': 'https://www.muyinteresante.es/tecnologia/articulo/quien-esta-detras-de-los-ciberataques-a-ucrania-521650353549',
    'urlToImage': 'https://estaticos.muyinteresante.es/uploads/images/article/625e65b45cafe89c7a65a1e4/19.jpg',
    'publishedAt': '2022-04-19T07:41:10Z',
    'content': 'Ucrania se ha visto afectada por un aluvión de ataques cibernéticos, que según su gobierno están en un nivel completamente diferente a los que se habían detectado hasta el momento. Hace unos días, an… [+2630 chars]',
  }],
}
