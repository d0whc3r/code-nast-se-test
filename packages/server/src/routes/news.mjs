import fetch from 'node-fetch'
import { generateUrl } from '../common.mjs'

const PAGE_SIZE = 10

/**
 * Wrapper function to get news and return in http response
 * @param params {Record<string, string>}
 * @param res {import('http').ServerResponse}
 */
async function getNews(params, res) {
  const page = +params.page || 1
  const pageSize = +params.pageSize || PAGE_SIZE
  const url = generateUrl({ ...params, pageSize: PAGE_SIZE })
  return fetch(url)
    .then((response) => response.json())
    .then((data) => {
      if (data.status === 'ok') {
        res.status(200).json({ ...data, pageSize, page, totalPages: Math.ceil(data.totalResults / PAGE_SIZE) })
      } else {
        res.status(500).json({ error: data.message })
      }
    })
    .catch((error) => {
      res.status(500).json({ error })
    })
}

/**
 * Get a list of news by topic
 * @param req {import('http').IncomingMessage}
 * @param res {import('http').ServerResponse}
 * @param next {import('express').NextFunction}
 */
export async function newsByTopic(req, res, next) {
  const params = { ...req.params, ...req.query }
  return getNews(params, res)
}

/**
 * Get a list of news by topic
 * @param req {import('http').IncomingMessage}
 * @param res {import('http').ServerResponse}
 * @param next {import('express').NextFunction}
 */
export async function latestNews(req, res, next) {
  const today = new Date().toISOString().split('T')[0]
  const yesterday = new Date(new Date().setDate(new Date().getDate() - 1)).toISOString().split('T')[0]
  const params = { sources: 'engadget', from: yesterday, to: today, ...req.query }
  return getNews(params, res)
}
