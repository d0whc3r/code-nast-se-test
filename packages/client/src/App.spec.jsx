import { describe, expect, it, vi } from 'vitest'
import { getNews } from './api/index'
import { mockedNews } from './api/news.mock'
import App from './App'
import { renderWithPagination, screen } from '../test/utils'

vi.mock('./api', () => {
  return {
    getNews: vi.fn(),
  }
})

describe('Simple working test', () => {
  it('the title is visible', () => {
    getNews.mockResolvedValue(mockedNews)
    const { baseElement } = renderWithPagination(<App />)
    expect(screen.getByText(/News board/i)).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })
})
