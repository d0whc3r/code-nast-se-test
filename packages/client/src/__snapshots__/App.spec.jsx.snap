// Vitest Snapshot v1

exports[`Simple working test > the title is visible 1`] = `
<body>
  <div>
    <header>
      <h1
        class="text-center"
      >
        News board
      </h1>
    </header>
    <main
      class="md:container md:mx-auto"
    >
      <div
        class="w-full"
      >
        <section
          class="md:sticky md:inset-x-0 md:top-0 md:left-0 md:bg-white md:z-20 md:h-44 px-8"
        >
          <form
            class="flex flex-col pt-6 bg-white rounded"
          >
            <div
              class="md:grid md:grid-cols-2"
            >
              <div
                class="input-container md:mr-4"
              >
                <label
                  class="label"
                  for="topic"
                >
                  Topic
                </label>
                <input
                  class="input"
                  id="topic"
                  name="topic"
                  placeholder="Topic to search"
                  type="text"
                />
              </div>
              <div
                class="md:grid md:grid-cols-2"
              >
                <div
                  class="input-container"
                >
                  <label
                    class="label"
                    for="from"
                  >
                    Start date
                  </label>
                  <input
                    class="input"
                    id="from"
                    name="from"
                    type="date"
                  />
                </div>
                <div
                  class="input-container"
                >
                  <label
                    class="label"
                    for="to"
                  >
                    End date
                  </label>
                  <input
                    class="input"
                    id="to"
                    name="to"
                    type="date"
                  />
                </div>
              </div>
            </div>
            <button
              class="button self-start"
              type="submit"
            >
              Search
            </button>
          </form>
        </section>
        <div
          class="px-8"
        >
          <div
            data-testid="loading"
          >
            <div
              class="w-full flex justify-center text-center m-10"
            >
              <svg
                class="w-8 h-8 mr-2 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600"
                fill="none"
                viewBox="0 0 100 101"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M100 50.59c0 27.615-22.386 50.001-50 50.001s-50-22.386-50-50 22.386-50 50-50 50 22.386 50 50Zm-90.919 0c0 22.6 18.32 40.92 40.919 40.92 22.599 0 40.919-18.32 40.919-40.92 0-22.598-18.32-40.918-40.919-40.918-22.599 0-40.919 18.32-40.919 40.919Z"
                  fill="currentColor"
                />
                <path
                  d="M93.968 39.04c2.425-.636 3.894-3.128 3.04-5.486A50 50 0 0 0 41.735 1.279c-2.474.414-3.922 2.919-3.285 5.344.637 2.426 3.12 3.849 5.6 3.484a40.916 40.916 0 0 1 44.131 25.769c.902 2.34 3.361 3.802 5.787 3.165Z"
                  fill="currentFill"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</body>
`;
